File: Medium-Array/Printmaxsubarray.java
import java.util.Arrays;
import java.util.Scanner;

public class Printmaxsubarray 
{
    public static int[] maxSubarraySum(int[] arr)
    {
        int sum = 0;
        int maxsum = Integer.MIN_VALUE;
        int start = 0;
        int tempStart = 0;
        int end = 0;
        
        for (int i = 0; i < arr.length; i++)
        {
            sum += arr[i];
            if (sum > maxsum) 
            {
                maxsum = sum;
                start = tempStart;
                end = i;
            }

            if (sum < 0)
            {
                sum = 0;
                tempStart = i + 1;
            }
        }

        // Extract the maximum subarray
        int[] maxSubarray = new int[end - start + 1];
        for (int i = start; i <= end; i++)
        {
            maxSubarray[i - start] = arr[i];
        }
        return maxSubarray;
    }

    public static void main(String[] args)
    {
       Scanner sc=new Scanner(System.in);
       System.out.println("Size of the array");
       int n=sc.nextInt();
       int[] arr=new int[n];
       System.out.println("Enter the elements of the array");
       for(int i=0;i<n;i++)
       {
        arr[i]=sc.nextInt();
       }
       System.out.println("Print Maximum subarray is "+Arrays.toString(maxSubarraySum(arr)));
    }
    
}

File: Medium-Array/Printmaxsubarray.py
# WAP to to print the maximum subarray from the given array.

def max_subarray(arr):
    max_sum = float('-inf')
    current_sum = 0
    start = 0
    end = 0
    temp_start = 0

    for i in range(len(arr)):
        current_sum += arr[i]

        if current_sum > max_sum:
            max_sum = current_sum
            start = temp_start
            end = i

        if current_sum < 0:
            current_sum = 0
            temp_start = i + 1

    return arr[start:end + 1]

arr=[]
n=int(input("Enter the number of elements in the array: "))
print("Enter the elements of the array:")
for i in range(n):
    element=int(input())
    arr.append(element)
result = max_subarray(arr)
print("The maximum subarray is:", result)   